src/node_crypto.cc

```diff
@@ -1904,25 +1930,17 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
                                 String::kNormalString, mem->length));
   BIO_free_all(bio);
 
-  unsigned int md_size, i;
   unsigned char md[EVP_MAX_MD_SIZE];
+  unsigned int md_size;
+  char fingerprint[EVP_MAX_MD_SIZE * 3];
```

Comment:
Since you're here, can you change that to `EVP_MAX_MD_SIZE * 3 + 1`?  There is a (hypothetical) write-after-end in case `md_size == EVP_MAX_MD_SIZE`.  (Hypothetical because that can only happen with SHA-512.)